File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
private:
vector<int> generateRow(int row) {
    long long ans = 1;
    vector<int> ansRow;
    ansRow.push_back(1); //inserting the 1st element

    //calculate the rest of the elements:
    for (int col = 1; col < row; col++) {
        ans = ans * (row - col);
        ans = ans / col;
        ansRow.push_back(ans);
    }
    return ansRow;
}
public:

    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> ans;
        // store the entire pascal's triangle:
        for (int row = 1; row <= numRows; row++) {
            ans.push_back(generateRow(row));
        }
        return ans;
    }
};

File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        result = []
        
        for rows in range(numRows):
            ans = 1
            ansRow = [1]
            for col in range(1,rows+1):
                ans = (ans * (rows - col+1)) // col
                ansRow.append(ans)
            result.append(ansRow)

        return result